### ISSUE TITLE
expecting T_PAAMAYIM_NEKUDOTAYIM in class-ep-config.php on line 80

### ISSUE THREAD

**Issue by celeph** [10up/ElasticPress#581]
Hi, I tried to update ElasticPress today but the plugin couldn't be activated due to a syntax error, unexpected ')', expecting T_PAAMAYIM_NEKUDOTAYIM  class-ep-config.php on line 80

This is a bit of a misleading error message, but it turns out this error is thrown because empty() does only support variables (not constants or other expressions) in PHP versions prior to 5.5. I found similar errors in other files, e.g. class-ep-dashboard.php (line 158). 

Easy to fix, but I'm not sure how many other lines would need to be changed, and whether it's worth supporting older versions at this point. I think if you could add a note to the readme, description, or changelog that the plugin requires PHP 5.5+ may help avoiding some surprises/confusion.

**Comment by tlovett1** [comment]
I'll make sure this gets fixed today. We are still supporting PHP 5.2.

**Comment by palpalani** [comment]
modules/woocommerce/woocommerce.php line number 351. also needs to fix.

**Comment by tlovett1** [comment]
Fixed. Any other errors?

**Comment by celeph** [comment]
Thanks! When I searched my workspace for similar cases I found one on line 799 of elasticpress/bin/wp-cli.php:  if ( empty( ep_get_host() ) ) 

I updated all 4 files and was able to activate the plugin without errors (using 5.3.29).

**Comment by tlovett1** [comment]
@celeph updated. Can you confirm that the version on Github works for PHP 5.2?

**Comment by celeph** [comment]
I don't have PHP 5.2 available at the moment, but the plugin is working with PHP 5.3.29. Tested successfully (in WP) using your updates in the development branch. WP-CLI is still throwing an error on line 799 though. It doesn't look like the change to wp-cli.php made it into the branch yet.

**Comment by tlovett1** [comment]
Just fixed.

**Comment by celeph** [comment]
thanks, wp-cli confirmed to work with PHP 5.3.29 as well

**Comment by tlovett1** [comment]
@celeph just prepped the version for release. Can you confirm one more time that all is good?

**Comment by celeph** [comment]
looking good, I just pulled the latest from develop, sync'd index, ran a few wp-cli commands, search works. No problems with PHP 5.3.  
I found another issue when I activated the admin-module. With admin-module active, post search results appear to be the same (all posts) regardless of the keyword used. I'm sure this is unrelated to the original PHP 5.3 issues, and may also be something specific to my config. Will test this out further tomorrow.
